kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/testplugins/ConstantSaslFilter.java

```diff
@@ -28,6 +32,11 @@ public ConstantSaslFilter(ConstantSasl.Config config) {
 
     @Override
     public CompletionStage<RequestFilterResult> onRequest(ApiKeys apiKey, RequestHeaderData header, ApiMessage request, FilterContext context) {
+        if (apiKey == ApiKeys.SASL_HANDSHAKE
+                || apiKey == ApiKeys.SASL_AUTHENTICATE) {
+            LOGGER.error("{} is not compatible with clients using SASL authentication. Disconnecting!", getClass().getName());
+            return context.requestFilterResultBuilder().drop().completed();
```

Comment:
I realise the API doc is wrong on drop().  The intent of drop is to drop the request - which only makes sense in case where the client expects no response (i.e. ProduceRequest with no-acks). 


In this case I think we want:

context.requestFilterResultBuilder().errorResponse(...).withCloseConnection().completed()
 
which will send the error response and close the connection.

I'll raise a PR to fix drop's Javadoc.